lib/compress/zstd_compress.c

```diff
@@ -2493,12 +2496,18 @@ static size_t ZSTD_compressBlock_targetCBlockSize(ZSTD_CCtx* zc,
                     srcSize,
                     zc->entropyWorkspace, HUF_WORKSPACE_SIZE /* statically allocated in resetCCtx */,
                     zc->bmi2);
-            if (!ZSTD_isError(cSize) && cSize != 0) {
+
+            if (cSize == 0) {
+                /* If compressSequences didn't work, we just output a regular
+                 * uncompressed block */
+                cSize = ZSTD_noCompressBlock(dst, dstCapacity, src, srcSize, lastBlock);
+                FORWARD_IF_ERROR(cSize);
+            } else {
                 U32 const cBlockHeader24 = lastBlock + (((U32)bt_compressed)<<1) + (U32)(cSize << 3);
                 MEM_writeLE24(ostart, cBlockHeader24);
                 cSize += ZSTD_blockHeaderSize;
             }
-        }
+        } else return cSize;
```

Comment:
Please reverse the if condition. `if (!blah) return cSize;`

A common pattern is to have "guards" or "short-circuits" that return if a condition isn't met.  In this function we want to return as soon as we hit an error we can't recover from. This keeps the nesting level down, and keeps the control flow linear, which is easy to read.